addons/mail/static/tests/mock_server/mock_models/res_partner.js

```diff
@@ -185,10 +185,16 @@ export class ResPartner extends webModels.ResPartner {
     }
 
     compute_im_status(partner) {
+        if (partner.im_status) {
+            return partner.im_status;
+        }
         if (partner.id === serverState.odoobotId) {
             return "bot";
         }
-        return partner.im_status;
+        if (!partner.user_ids.length) {
+            return "im_status";
+        }
```

Comment:
Did you mean sentinel value `"im_partner"`?
Note that modeling can be incomplete in JS by design, and `"im_partner"` value means we don't fetch it anymore because we know it's not related to any user (it's a pure partner). I don't think this must be computed in JS that way, instead the fetch data should return `"im_partner"` value.

Reply:
Oh I'm stupid, I thought this was a JS model, not mock server model. It's ok then, but I still think you meant `"im_partner"` sentinel value.